lib/ansible/modules/cloud/vmware/vmware_host_dns_facts.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2018, Abhijeet Kasurde <akasurde@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r'''
+---
+module: vmware_host_dns_facts
+short_description: Gathers facts about an ESXi host's DNS configuration information
+description:
+- This module can be used to gather facts about an ESXi host's DNS configuration information when ESXi hostname or Cluster name is given.
+- All parameters and VMware object names are case sensitive.
+version_added: '2.5'
+author:
+- Abhijeet Kasurde (@akasurde)
+notes:
+- Tested on vSphere 6.5
+requirements:
+- python >= 2.6
+- PyVmomi
+options:
+  cluster_name:
+    description:
+    - Name of the cluster from which the ESXi host belong to.
+    - If C(esxi_hostname) is not given, this parameter is required.
+  esxi_hostname:
+    description:
+    - ESXi hostname to gather facts from.
+    - If C(cluster_name) is not given, this parameter is required.
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Gather DNS facts about all ESXi Host in given Cluster
+  vmware_host_dns_facts:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    cluster_name: cluster_name
+
+- name: Gather DNS facts about ESXi Host
+  vmware_host_dns_facts:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+'''
+
+RETURN = r'''
+drs_rule_facts:
+    description: metadata about DNS config from given cluster / host system
+    returned: always
+    type: dict
+    sample: {
+                "DC0_C0_H0": {
+                    "dhcp": true,
+                    "domain_name": "localdomain",
+                    "host_name": "localhost",
+                    "ip_address": [
+                        "8.8.8.8"
+                    ],
+                    "search_doamin": [
+                        "localdomain"
+                    ],
+                    "virtual_nic_device": "vmk0"
+                }
+            }
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import vmware_argument_spec, PyVmomi
+
+
+class VmwareDnsFactsManager(PyVmomi):
+    def __init__(self, module):
+        super(VmwareDnsFactsManager, self).__init__(module)
+        cluster_name = self.params.get('cluster_name', None)
+        esxi_host_name = self.params.get('esxi_hostname', None)
+        self.hosts = []
+        if cluster_name:
```

Comment:
As this piece of code is being added to many modules (getting hosts), perhaps we can put that in the vmware module utils at some point? (no need now, just an improvement for the future :))

Reply:
Make sense. I will add it in next PR. This thought came in my mind as well. But I was too lazy to do it. anyways. ;)